packages/common/src/elevate.ts

```diff
@@ -33,21 +45,33 @@ function createShadow({ level, color, depth, spread }) {
   return shadows.slice(0, depth).join();
 }
 
-function elevate({
+interface ElevateProps {
+  color?: string;
+  level?: number;
+  depth?: number;
+  spread?: boolean;
+  fallback?: boolean;
+}
+
+function  elevate({
   color = '#000',
   level,
   depth = 3,
   spread = true,
   fallback = true,
-}) {
+}: ElevateProps) {
   const all = [
     fallback ? '0 0 0 rgba(0, 0, 0, 0)' : 'none',
     createShadow({ level: 0, color, depth, spread }),
     createShadow({ level: 1, color, depth, spread }),
     createShadow({ level: 2, color, depth, spread }),
   ];
 
-  return all[level] || all;
+  if (level) {
+    return all[level];
+  }
+
+  return all;
```

Comment:
The previous way seemed more clear, i would only using nullish coalescing operator. WDYT?
```suggestion
  all[level] ?? all;
```

Reply:
Hey, @MicaelRodrigues!
I don't think the previous way is more clear. Someone new to JS may have difficulty to understand `all[level] ?? all`. Contrary to what you said, I believe that this new implementation is easier to read and to understand the code.

Don't you think it's easier to read the statements if we were reading like it was an English text, instead of a code full of JS patterns and operators?

Reply:
honestly, I understand both points but there's two things that seem more important to me here:
- what if I give it a higher level? Right now, as we accept any number I could give it a 500 and would get an undefined. Should we create the array based on the given level?
- this function signature is, IMO, weird. It can either return an array or a value. I'm not the biggest fan of doing this as it makes the usage a lot more complex. Maybe we should type the return differently based on the fact that a level is given or not using [function overloads](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads)

Reply:
> Hey, @MicaelRodrigues! I don't think the previous way is more clear. Someone new to JS may have difficulty to understand `all[level] ?? all`. Contrary to what you said, I believe that this new implementation is easier to read and to understand the code.
> 
> Don't you think it's easier to read the statements if we were reading like it was an English text, instead of a code full of JS patterns and operators?

> * could give it a 500 and would get an undefined. Should we create the arr

It's a question of flavour :) We expect people working in this code to know javascript fundamentals. If they don't we are here to help improve the code... :)
I agree with you in "as it was an English text" and both options read plain english to me (ex: my suggestion reads ("a" or "b"). 

Anyway, the important concerns in this function are the ones @goncalo-matos exposed ;)

Reply:
Can we use the value "0" as `level`? If so the current implementation will ignore the level when it is "0".